### DISCUSSION TITLE
Mobile web cersion UI is messed up

### DISCUSSION THREAD

**Discussion by mnobeidat13** [flet-dev/flet#600] [Q&A]
I deployed my app on repl.it. The interface renders fine when I open the app from my laptop but not so good when using my phone
On laptop:
![image](https://user-images.githubusercontent.com/32016587/202868196-7056d7dc-fafb-40bc-a3e4-2413f282ab36.png)

On mobile:
![image](https://user-images.githubusercontent.com/32016587/202868221-d2de6adb-18ce-4c1c-aefb-7dd9952050e0.png)

What would the problem be?

**Comment by ndonkoHenri** [comment]
I think your phone's browser (or any other browser) is just rendering AS IS. That means, you built it that way, and all browsers will render it that way.
I don't know what you expected from the mobile version particularly, but when building your apps, and willing to test how it could behave on mobile, do this:
- if you are developing using your desktop's browser to see the results, resize the browser's window to common phone sizes and observe the result.
- if you are using a native desktop window, then you should also try resizing it to common phone sizes and observe the result.

There are some controls/properties you can use to ensure your stuffs fit the device's window: 
- controls: [ResponsiveRow](https://flet.dev/docs/controls/responsiverow),
- properties: [Row.wrap](https://flet.dev/docs/controls/row#wrap), [Column.wrap](https://flet.dev/docs/controls/column#wrap)

**Reply by mnobeidat13** [reply]
Well, I was not aware the UI would not fit both platforms.  
I cannot get ResponsiveRow working with my controls.
Is there a way to read the type of device the user is using to be able to render a compatible version for each device?

**Reply by ndonkoHenri** [reply]
Yes there is:

https://flet.dev/docs/controls/page#platform

https://flet.dev/docs/controls/page#web

https://flet.dev/docs/controls/page#pwa

**Reply by mnobeidat13** [reply]
Thanks man. It works fine. But page.platform detects iphone as macos not ios. will do more testing.
